src/mds/Server.cc

```diff
@@ -5722,6 +5722,43 @@ int Server::parse_layout_vxattr(string name, string value, const OSDMap& osdmap,
   return 0;
 }
 
+int64_t Server::return_bytes(string& value_str)
+{
+  if (value_str.empty()) {
+    return -CEPHFS_EINVAL;
+  }
+  unordered_map<char, int64_t> size_values = {{'k', 1L << 10}, {'m', 1L << 20}, 
+                                             {'g', 1L << 30}, {'t', 1L << 40}};
+  char data_size = tolower(value_str.back());
+  string value = value_str.substr(0, value_str.length() - 1);
+  if (size_values.find(data_size) != size_values.end()) {
+    try {
+      return (boost::lexical_cast<int64_t>(value)) * size_values[data_size];
+    } catch (boost::bad_lexical_cast const&) {
+      if(value.empty()) {
+        dout(10) << "cannot cast single character alphabetic value: " 
+        << value_str << dendl;
+      } else {
+        dout(10) << "invalid numerical value encountered: " << value << dendl;
```

Comment:
nit: probably reword log message as ...
`"bad coefficient: '" << value << "' for '" << last_char << "' size spec"`

Reply:
fixed.